# Fitness Trackers Products Ecommerce

Download link : https://www.kaggle.com/datasets/arnabchaki/fitness-trackers-products-ecommerce/data

## About Data
This is a fitness tracker product dataset consisting of different products from various brands with their specifications, ratings and reviews for the Indian market. The data has been collected from e-commerce websites namely Flipkart and Amazon using web scraping technique.

Inspiration

This dataset could be used to find answers to some interesting questions like -

* Is there a significant demand for fitness trackers in the Indian market?
* Information on the top 5 brands for fitness bands and smart watches
* Is there a correlation between the prices and product specifications, ratings, etc.
* Different types of fitness trackers, their price segments for different users

This dataset contains 451 samples with 16 attributes. There are some missing values in this dataset.

## 1. Data Checks to perform

* Check data type
* Check Duplicates
* Check Missing values
* Check the number of unique values of each column and various categories present in the different column
* Check statistics of data set

#### Import libraries and read dataset

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
df = pd.read_csv("smartwatches.csv")

In [3]:
df.head()

Unnamed: 0       Brand  Current Price  Original Price  Discount Percentage  \
0           0       noise        82990.0         89900.0             7.686318   
1           1  fire-boltt         3799.0         16999.0            77.651627   
2           2        boat         1999.0          7990.0            74.981227   
3           3  fire-boltt         1799.0         19999.0            91.004550   
4           4       noise         1599.0          4999.0            68.013603   

   Rating  Number OF Ratings                              Model Name  \
0     4.0               65.0   Wrb-sw-colorfitpro4alpha-std-rgld_pnk   
1     4.3            20788.0                                  BSW046   
2     3.8            21724.0                               Wave Call   
3     4.3            13244.0                                  BSW053   
4     4.1            13901.0  Wrb-sw-colorfitpulsegobuzz-std-blk_blk   

  Dial Shape Strap Color Strap Material Touchscreen  Battery Life (Days)  \
0        NaN         NaN            NaN         NaN                  8.0   
1        NaN         NaN        Silicon         Yes                  3.5   
2        NaN         NaN        Silicon         Yes                  8.0   
3        NaN         NaN        Silicon         Yes                  3.5   
4        NaN         NaN          Other         Yes                  8.0   

  Bluetooth Display Size     Weight  
0       Yes          NaN  35 - 50 g  
1       Yes   1.8 inches  50 - 75 g  
2       Yes   1.7 inches  35 - 50 g  
3       Yes   1.8 inches      75g +  
4       Yes   1.7 inches  35 - 50 g

In [4]:
df.shape

(450, 16)

Insight
* 450 rows and 16 columns

#### Dataset info

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 450 entries, 0 to 449
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Unnamed: 0           450 non-null    int64  
 1   Brand                450 non-null    object 
 2   Current Price        440 non-null    float64
 3   Original Price       377 non-null    float64
 4   Discount Percentage  377 non-null    float64
 5   Rating               444 non-null    float64
 6   Number OF Ratings    391 non-null    float64
 7   Model Name           415 non-null    object 
 8   Dial Shape           327 non-null    object 
 9   Strap Color          327 non-null    object 
 10  Strap Material       381 non-null    object 
 11  Touchscreen          413 non-null    object 
 12  Battery Life (Days)  418 non-null    float64
 13  Bluetooth            444 non-null    object 
 14  Display Size         420 non-null    object 
 15  Weight               264 non-null    obj

* Maximum columns have missing value
* 7 columns are numeric and 9 columns are categorical

In [6]:
df["Unnamed: 0"].unique()

array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,
        26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,
        39,  40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,
        52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,
        65,  66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,
        78,  79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,
        91,  92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103,
       104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116,
       117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129,
       130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142,
       143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155,
       156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168,
       169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 18

In [7]:
len(df["Unnamed: 0"].unique())

450

Unnamed: 0 and index are same

**Insights**
* Maximum columns have missing value
* 7 columns are numeric and 9 columns are categorical
* Unnamed: 0 and index are same

#### Droping "Unnamed: 0" column

In [8]:
df.drop("Unnamed: 0",axis=1,inplace=True)

In [9]:
df.head()

Brand  Current Price  Original Price  Discount Percentage  Rating  \
0       noise        82990.0         89900.0             7.686318     4.0   
1  fire-boltt         3799.0         16999.0            77.651627     4.3   
2        boat         1999.0          7990.0            74.981227     3.8   
3  fire-boltt         1799.0         19999.0            91.004550     4.3   
4       noise         1599.0          4999.0            68.013603     4.1   

   Number OF Ratings                              Model Name Dial Shape  \
0               65.0   Wrb-sw-colorfitpro4alpha-std-rgld_pnk        NaN   
1            20788.0                                  BSW046        NaN   
2            21724.0                               Wave Call        NaN   
3            13244.0                                  BSW053        NaN   
4            13901.0  Wrb-sw-colorfitpulsegobuzz-std-blk_blk        NaN   

  Strap Color Strap Material Touchscreen  Battery Life (Days) Bluetooth  \
0         NaN            NaN         NaN                  8.0       Yes   
1         NaN        Silicon         Yes                  3.5       Yes   
2         NaN        Silicon         Yes                  8.0       Yes   
3         NaN        Silicon         Yes                  3.5       Yes   
4         NaN          Other         Yes                  8.0       Yes   

  Display Size     Weight  
0          NaN  35 - 50 g  
1   1.8 inches  50 - 75 g  
2   1.7 inches  35 - 50 g  
3   1.8 inches      75g +  
4   1.7 inches  35 - 50 g

#### Check duplicates row and column

Check duplicate row

In [10]:
df.duplicated().sum()

10

There is no duplicate row

Check duplicate column

In [11]:
def duplicate_cols(data):
    dup_cols = set()
    for x in range(data.shape[1]):
        base_col = data.iloc[:,x]
        for y in range(x+1,data.shape[1]):
            comp_col = data.iloc[:,y]
            if base_col.equals(comp_col):
                dup_cols.add(data.columns.values[y])
    return dup_cols
duplicate_cols(df)

set()

There is no duplicate column

**Insights**
* There is no duplicate row and column

#### Check missing value

In [12]:
df.isnull().sum()

Brand                    0
Current Price           10
Original Price          73
Discount Percentage     73
Rating                   6
Number OF Ratings       59
Model Name              35
Dial Shape             123
Strap Color            123
Strap Material          69
Touchscreen             37
Battery Life (Days)     32
Bluetooth                6
Display Size            30
Weight                 186
dtype: int64

In [13]:
df[df.isnull().any(axis=1)]

Brand  Current Price  Original Price  Discount Percentage  Rating  \
0         noise        82990.0         89900.0             7.686318     4.0   
1    fire-boltt         3799.0         16999.0            77.651627     4.3   
2          boat         1999.0          7990.0            74.981227     3.8   
3    fire-boltt         1799.0         19999.0            91.004550     4.3   
4         noise         1599.0          4999.0            68.013603     4.1   
..          ...            ...             ...                  ...     ...   
445  fire-boltt         5999.0          8999.0            33.337037     2.5   
446  fire-boltt         3700.0          9999.0            62.996300     NaN   
447  fire-boltt         5999.0          8999.0            33.337037     2.5   
448  fire-boltt         3200.0          7999.0            59.994999     NaN   
449  fire-boltt         3000.0          7999.0            62.495312     NaN   

     Number OF Ratings                              Model Name Dial Shape  \
0                 65.0   Wrb-sw-colorfitpro4alpha-std-rgld_pnk        NaN   
1              20788.0                                  BSW046        NaN   
2              21724.0                               Wave Call        NaN   
3              13244.0                                  BSW053        NaN   
4              13901.0  Wrb-sw-colorfitpulsegobuzz-std-blk_blk        NaN   
..                 ...                                     ...        ...   
445                NaN                                  bsw003     Circle   
446                NaN                               beast pro     Square   
447                NaN                                  bsw003     Circle   
448                NaN                                  call 2     Square   
449                NaN                                  call 2     Square   

    Strap Color Strap Material Touchscreen  Battery Life (Days) Bluetooth  \
0           NaN            NaN         NaN                  8.0       Yes   
1           NaN        Silicon         Yes                  3.5       Yes   
2           NaN        Silicon         Yes                  8.0       Yes   
3           NaN        Silicon         Yes                  3.5       Yes   
4           NaN          Other         Yes                  8.0       Yes   
..          ...            ...         ...                  ...       ...   
445        Grey        Silicon         Yes                 22.0       Yes   
446       Black        Silicon         Yes                  8.0       Yes   
447        Gold        Silicon         Yes                 22.0       Yes   
448        Blue        Silicon         Yes                 22.0       Yes   
449       Black        Silicon         Yes                 22.0       Yes   

    Display Size     Weight  
0            NaN  35 - 50 g  
1     1.8 inches  50 - 75 g  
2     1.7 inches  35 - 50 g  
3     1.8 inches      75g +  
4     1.7 inches  35 - 50 g  
..           ...        ...  
445   1.3 inches        NaN  
446   0.1 inches        NaN  
447   1.3 inches        NaN  
448   0.1 inches        NaN  
449   0.1 inches        NaN  

[348 rows x 15 columns]

**Insight**
* Among 450 rows, 348 rows have missing values in a single or multiple columns

#### Differentiate the categorical and Numerical Columns

In [16]:
categorical_columns = [feature for feature in df.columns if df[feature].dtype == "O"]
numerical_columns = [feature for feature in df.columns if df[feature].dtype != "O"]

In [17]:
print("Categorical features : {}".format(categorical_columns))
print("Numerical features : {}".format(numerical_columns))

Categorical features : ['Brand', 'Model Name', 'Dial Shape', 'Strap Color', 'Strap Material', 'Touchscreen', 'Bluetooth', 'Display Size', 'Weight']
Numerical features : ['Current Price', 'Original Price', 'Discount Percentage', 'Rating', 'Number OF Ratings', 'Battery Life (Days)']


#### Check the unique values in the columns

In [21]:
# Categorical feature unique no
for feature in categorical_columns:
    lenght_of_unique_values = len(df[feature].unique())
    print("'{0}' feature's no of unique values number : '{1}'".format(feature,lenght_of_unique_values))

'Brand' feature's no of unique values number : '18'
'Model Name' feature's no of unique values number : '390'
'Dial Shape' feature's no of unique values number : '7'
'Strap Color' feature's no of unique values number : '31'
'Strap Material' feature's no of unique values number : '16'
'Touchscreen' feature's no of unique values number : '3'
'Bluetooth' feature's no of unique values number : '3'
'Display Size' feature's no of unique values number : '41'
'Weight' feature's no of unique values number : '6'


In [22]:
# Numerical feature unique no
for feature in numerical_columns:
    lenght_of_unique_values = len(df[feature].unique())
    print("'{0}' feature's no of unique values number : '{1}'".format(feature,lenght_of_unique_values))

'Current Price' feature's no of unique values number : '165'
'Original Price' feature's no of unique values number : '105'
'Discount Percentage' feature's no of unique values number : '272'
'Rating' feature's no of unique values number : '28'
'Number OF Ratings' feature's no of unique values number : '251'
'Battery Life (Days)' feature's no of unique values number : '8'


In [32]:
#for showing 390 rows
pd.set_option("display.max_rows",390)

df['Model Name'].value_counts()

Model Name
beast pro                                                                           5
ring                                                                                5
Wave Call                                                                           3
Watch Flash                                                                         3
BSW024                                                                              3
starcross                                                                           2
BSW049                                                                              2
BSW030                                                                              2
BSW042                                                                              2
RING 3                                                                              2
NINJA PRO MAX                                                                       2
8.90E+12                                   